x/spec/README.md

```diff
@@ -0,0 +1,281 @@
+# `x/spec`
+
+## Abstract
+
+This document specifies the spec module of Lava Protocol.
+
+The spec module is responsible for managing Lava's chains specifications. a spec is a struct difining all the api's of a specific chain and its bahaviours. The specs determines the chain properties like block time, finalization distance, imports and more. Adding a spec is done with a proposal on chain with the gov module. The first step of providers and subscriptions to work in lava is having a specification for the wanted chain.
+
+This document is focuses on the spec' technical aspects. For more information on chain support in lava see pairing README.
+
+## Contents
+* [Concepts](#concepts)
+  * [Spec](#spec)
+  * [ApiCollections](#apicollections)
+  * [CollectionData](#collectiondata)
+  * [ParseDirective](#parsedirective)
+  * [Verification](#verification)
+  * [Import](#import)
+* [Parameters](#parameters)
+* [Queries](#queries)
+* [Transactions](#transactions)
+* [Proposals](#proposals)
+
+## Concepts
+
+### Spec
+
+A chain spec consists of general properties of the chain and a list of interface it supports. To make specs simple to create and maintain specs can import api's from another, for example, X testnet spec can Import from X mainnet spec and thus not need to redifine all of the interfaces.
```

Comment:
AI:

A chain spec consists of general properties of the chain and a list of interfaces it supports. To simplify the creation and maintenance of specs, they can import APIs from another spec. For example, the X testnet spec can import from the X mainnet spec, eliminating the need to redefine all of the interfaces.